tests/test_sub_port_intf.py

```diff
@@ -851,23 +915,32 @@ def _test_sub_port_intf_removal(self, dvs, sub_port_intf_name, removal_seq_test=
         # Verify that sub port router interface entry is removed from ASIC_DB
         self.check_sub_port_intf_key_removal(self.asic_db, ASIC_RIF_TABLE, rif_oid)
 
-        # Verify physical port host interface vlan tag attribute
-        fv_dict = {
-            "SAI_HOSTIF_ATTR_VLAN_TAG": "SAI_HOSTIF_VLAN_TAG_STRIP",
-        }
-        for phy_port in phy_ports:
-            hostif_oid = dvs.asicdb.hostifnamemap[phy_port]
-            self.check_sub_port_intf_fvs(self.asic_db, ASIC_HOSTIF_TABLE, hostif_oid, fv_dict)
+        if removal_seq_test == True:
+            # Verify that parent port is removed from ASIC_DB
+            self.asic_db.wait_for_n_keys(asic_tbl_name, obj_cnt - 1)
```

Comment:
Can you check L897? for the logic to work from 898 onwards, that has to be outside the if condition? If that is true, then wouldn't this be obj_cnt - 2? (parent port and subport gone)

Reply:
Parent port ("ASIC_STATE:SAI_OBJECT_TYPE_PORT" for physical port, "ASIC_STATE:SAI_OBJECT_TYPE_LAG" for LAG) and sub port ("ASIC_STATE:SAI_OBJECT_TYPE_ROUTER_INTERFACE") are different SAI object types, hence in different ASIC_DB tables. Here, we check parent port count from asic_tbl_name, which refers to "ASIC_STATE:SAI_OBJECT_TYPE_PORT" or "ASIC_STATE:SAI_OBJECT_TYPE_LAG" depending on the parent port type.

Count check for sub port in ASIC_DB is on line 898.

Reply:
Thanks for clarification on the obj count. For removal_seq_test ==True, L897 will not get executed. so subport still exist. Checks on L898 and onwards are for checking if subport has been removed. Am I missing something here?

Reply:
The sub port removal call for `removal_seq_test == True` case is on earlier L846, which is not part of this PR. Because ip address still exists on sub port, sub port is not removed from ASIC_DB until ip address removal call is issued on L888. If `removal_seq_test == False` case, removal is in the correct order, i.e., ip address on a sub port is removed before sub port removal itself.